contivctl/contivctl/main.go

```diff
@@ -0,0 +1,195 @@
+package main
+
+import (
+	"os"
+
+	"github.com/codegangsta/cli"
+	"github.com/contiv/netplugin/contivctl"
+)
+
+var tenantFlag = cli.StringFlag{
+	Name:  "tenant, t",
+	Value: "default",
+	Usage: "Name of the tenant to create the policy on.",
+}
+
+func main() {
+	app := cli.NewApp()
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:   "master",
+			Value:  contivctl.DefaultMaster,
+			Usage:  "The hostname of the netmaster",
+			EnvVar: "NETMASTER",
+		},
+	}
+	app.Commands = []cli.Command{
+		{
+			Name:  "group",
+			Usage: "Endpoint Group manipulation tools",
+			Subcommands: []cli.Command{
+				{
+					Name:  "create",
+					Usage: "Create an endpoint group",
+					Flags: []cli.Flag{
+						tenantFlag,
+						cli.StringFlag{
+							Name:  "policy, p",
+							Usage: "Policy List (separated by commas)",
+						},
+					},
+					Action: contivctl.CreateEndpointGroup,
+				},
+				{
+					Name:   "delete",
+					Usage:  "Delete an endpoint group",
+					Flags:  []cli.Flag{tenantFlag},
+					Action: contivctl.DeleteEndpointGroup,
+				},
+				{
+					Name:  "list",
+					Usage: "List endpoint groups",
+					Flags: append([]cli.Flag{tenantFlag},
+						cli.BoolFlag{
+							Name:  "all, a",
+							Usage: "List endpoint groups for all tenants",
+						}),
+					Action: contivctl.ListEndpointGroups,
+				},
+			},
+		},
+		{
+			Name:  "net",
+			Usage: "Network manipulation tools",
+			Subcommands: []cli.Command{
+				{
+					Name:  "list",
+					Usage: "List networks",
+					Flags: append([]cli.Flag{tenantFlag}, cli.BoolFlag{
+						Name:  "all, a",
+						Usage: "Display all networks",
+					}),
+					Action: contivctl.ListNetworks,
+				},
+				{
+					Name:   "delete",
+					Usage:  "Delete a network",
+					Flags:  []cli.Flag{tenantFlag},
+					Action: contivctl.DeleteNetwork,
+				},
+				{
+					Name:  "create",
+					Usage: "Create a network",
+					Flags: []cli.Flag{
+						tenantFlag,
+						cli.BoolFlag{
+							Name:  "public, p",
+							Usage: "Indicate this network is public",
+						},
+						cli.StringFlag{
+							Name:  "encap, e",
+							Usage: "Packet tag",
+							Value: "vxlan",
+						},
+						cli.StringFlag{
+							Name:  "subnet, s",
+							Usage: "Subnet CIDR - REQUIRED",
+						},
+						cli.StringFlag{
+							Name:  "default-gw, g",
+							Usage: "Default Gateway - REQUIRED",
+						},
+					},
+					Action: contivctl.CreateNetwork,
+				},
+			},
+		},
+		{
+			Name:  "rule",
+			Usage: "Rule manipulation tools",
+			Subcommands: []cli.Command{
+				{
+					Name:   "list",
+					Usage:  "List rules for a given tenant/policy",
+					Flags:  []cli.Flag{tenantFlag},
+					Action: contivctl.ListRules,
+				},
+				{
+					Name:   "delete",
+					Usage:  "Delete an existing rule.",
+					Flags:  []cli.Flag{tenantFlag},
+					Action: contivctl.DeleteRule,
+				},
+				{
+					Name:  "add",
```

Comment:
minor nit, should this sub-command be called `create` as well just like sub-commands for policy, network etc?